<launch>
  <arg name="OPENBLAS_NUM_THREADS" default="1" />
  <arg name="MKL_NUM_THREADS" default="1" />
  <arg name="NUMEXPR_NUM_THREADS" default="1" />
  <arg name="OMP_NUM_THREADS" default="1" />
  <env name='OPENBLAS_NUM_THREADS' value='$(arg OPENBLAS_NUM_THREADS)' />
  <env name='MKL_NUM_THREADS' value='$(arg MKL_NUM_THREADS)' />
  <env name='NUMEXPR_NUM_THREADS' value='$(arg NUMEXPR_NUM_THREADS)' />
  <env name='OMP_NUM_THREADS' value='$(arg OMP_NUM_THREADS)' />

  <arg name="server" default="localhost"/>
  <arg name="port" default="22"/>
  <arg name="user" default=""/>
  <arg name="gpu_device" default="0"/>
  <arg name="env_loader" default="~/.octopus/config/env_loaders/octopus.env.sh"/>
  <arg name="heartbeat_id" default="system_monitor"/>
  <arg name="server_b" default="False"/>

  <arg name="cam_id" default="1"/>
  <arg name="is_cam390" default="True"/>
  <arg name="vehicle_name" default="vehicle-default"/>

  <machine name="$(arg server)" address="$(arg server)" ssh-port="$(arg port)" user="$(arg user)" env-loader="$(arg env_loader)"/>
  
  <group>
    <node name="cam_intrinsic_calibrator" pkg="cam_intrinsic_calibrator" type="run_cam_intrinsic_calibrator.py" machine="$(arg server)">
      <param name="cam_id" value="$(arg cam_id)"/>
      <param name="is_cam390" value="$(arg is_cam390)"/>
      <param name="heartbeat_id" value="$(arg heartbeat_id)"/>
      <param name="vehicle_name" value="$(arg vehicle_name)"/>
    </node>
  </group>

</launch>